### PR TITLE
Context provider added

### PR DESCRIPTION
**Pull request by gsrathoreniks**

### CODE REVIEW

File: flare/src/main/AndroidManifest.xml [hunk 1]
```diff
@@ -6,4 +6,10 @@
     <uses-permission android:name="android.permission.ACCESS_NETWORK_STATE" />
     <uses-permission android:name="android.permission.INTERNET" />
     <uses-permission android:name="android.permission.WAKE_LOCK" />
+    <application>
+        <provider
+            android:authorities="io.pantheon.flare.FlareProvider"
+            android:name=".FlareProvider"
+            android:exported="false" />
+    </application>
 </manifest>
```
Comment (LaatonWalaBhoot) [inline flare/src/main/AndroidManifest.xml]: change this name to flare context provider

File: flare/src/main/java/io/pantheon/flare/adapters/FirebaseAnalyticsAdapter.kt [hunk 2]
```diff
@@ -13,26 +13,29 @@
...
+        context?.let { context: Context ->
             firebaseAnalytics = FirebaseAnalytics.getInstance(context)
+            block(firebaseAnalytics)
         }
-        block(firebaseAnalytics)
         return this
     }
 
     override fun logEvent(eventName: String, eventMap: HashMap<String?, Any?>) {
         require(::firebaseAnalytics.isInitialized) { Exception("Flare: FirebaseAnalytics not initialized") }
```
Comment (LaatonWalaBhoot) [inline flare/src/main/java/io/pantheon/flare/adapters/FirebaseAnalyticsAdapter.kt]: Can we shift this check to base class as well?

**Review by LaatonWalaBhoot — CHANGES_REQUESTED** [review]

File: flare/src/main/java/io/pantheon/flare/adapters/FirebaseAnalyticsAdapter.kt (hunk 2 above)
Comment (gsrathoreniks) [reply flare/src/main/java/io/pantheon/flare/adapters/FirebaseAnalyticsAdapter.kt]: Done!

File: flare/src/main/AndroidManifest.xml (hunk 1 above)
Comment (gsrathoreniks) [reply flare/src/main/AndroidManifest.xml]: Done

File: flare/src/main/java/io/pantheon/flare/adapters/CleverTapAnalyticsAdapter.kt [hunk 3]
```diff
@@ -13,18 +13,17 @@
...
+package io.pantheon.flare.adapters
 
 import com.clevertap.android.sdk.CleverTapAPI
 
 class CleverTapAnalyticsAdapter : AnalyticsAdapter<CleverTapAPI>() {
 
-    override fun initialize(block: CleverTapAPI?.() -> Unit): AnalyticsAdapter<CleverTapAPI> {
-        block(CleverTapAPI.getDefaultInstance(null))
-        return this
+    override fun initClient(): CleverTapAPI? {
```
Comment (gsrathoreniks) [inline flare/src/main/java/io/pantheon/flare/adapters/CleverTapAnalyticsAdapter.kt:22]: Nullable return type because , CT's instance can be nullable!

**Review by LaatonWalaBhoot — APPROVED** [review]

**Pull request merged**